dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/imports/ImportServiceBean.java

```diff
@@ -277,43 +275,34 @@ public Dataset doImportHarvestedDataset(DataverseRequest dataverseRequest, Harve
             }
 
             // Check data against validation constraints
             // If we are migrating and "scrub migration data" is true we attempt to fix invalid data
             // if the fix fails stop processing of this file by throwing exception
-            Set<ConstraintViolation> invalidViolations = ds.getVersions().get(0).validate();
-            ValidatorFactory factory = Validation.buildDefaultValidatorFactory();
-            Validator validator = factory.getValidator();
-            if (!invalidViolations.isEmpty()) {
-                for (ConstraintViolation<DatasetField> v : invalidViolations) {
-                    DatasetField f = v.getRootBean();
+            validationResults = fieldValidationService.validateFieldsOfDatasetVersion(versionToValidate);
+            if (!validationResults.isEmpty()) {
+                DatasetFieldValidationService.ValidatorsWithContext validators
+                        = fieldValidationService.createValidatorsWithContext(versionToValidate);
+                for (ValidationResult result : validationResults) {
+                    DatasetField field = result.getField();
                     boolean fixed = false;
-                    boolean converted = false;
-                    // TODO: Is this scrubbing something we want to continue doing? 
                     if (settingsService.isTrueForKey(SettingsServiceBean.Key.ScrubMigrationData)) {
-                        fixed = processMigrationValidationError(f, cleanupLog, metadataFile.getName());
-                        converted = true;
+                        fixed = processMigrationValidationError(field, cleanupLog, metadataFile.getName());
                         if (fixed) {
-                            Set<ConstraintViolation<DatasetField>> scrubbedViolations = validator.validate(f);
-                            if (!scrubbedViolations.isEmpty()) {
-                                fixed = false;
-                            }
+                            fixed = validators.validateField(field).isOk();
                         }
                     }
                     if (!fixed) {
-                        String msg = "Data modified - File: " + metadataFile.getName() + "; Field: " + f.getDatasetFieldType().getDisplayName() + "; "
-                                + "Invalid value:  '" + f.getValue() + "'" + " Converted Value:'" + DatasetField.NA_VALUE + "'";
-                        cleanupLog.println(msg);
-                        f.setFieldValue(DatasetField.NA_VALUE);
-
+                        String message = String.format("Data modified - File: %s; Field: %s; Invalid value:  '%s' Converted Value:'%s'",
+                                metadataFile.getName(), field.getDatasetFieldType().getDisplayName(), field.getValue(), DatasetField.NA_VALUE);
+                        cleanupLog.println(message);
+                        field.setFieldValue(DatasetField.NA_VALUE);
                     }
```

Comment:
I would try to rewrite the whole thing as follows:
```
DatasetVersion versionToValidate = ds.getVersions().get(0);
List<ValidationResult> validationResults = fieldValidationService.validateFieldsOfDatasetVersion(versionToValidate);

if (!validationResults.isEmpty()) {

if (settingsService.isTrueForKey(SettingsServiceBean.Key.ScrubMigrationData)) {
  validationResults.forEach(r -> processMigrationValidationError(r.getField(), cleanupLog, metadataFile.getName()));

  validationResults = fieldValidationService.validateFieldsOfDatasetVersion(versionToValidate);
  validationResults.forEach(r -> cleanupLog.println(String.format("Data modified - File: %s; Field: %s; Invalid value:  '%s' Converted Value:'%s'", metadataFile.getName(), field.getDatasetFieldType().getDisplayName(), field.getValue(), DatasetField.NA_VALUE)));
  validationResults.forEach(r -> r.getField().setFieldValue(DatasetField.NA_VALUE));
} else {
  validationResults.forEach(r -> r.getField().setFieldValue(DatasetField.NA_VALUE));
}

}
    
```
If we are making an assumption that one validation can depend one other field value then checking only values of fields that had failed to validate in the first run is not so accurate (`ScrubMigrationData` case).
That's why I would try to get rid of field by field validation and just run it on the all fields.